Bump Go version in prep for a release

milosgajdos opened the pull request:
This bumps the Go version in prep for the next release.

thaJeztah commented on .github/workflows/build.yml:
```diff
@@ -33,8 +33,8 @@ jobs:
         go:
-          - 1.22.12
-          - 1.23.6
+          - 1.23.7
+          - 1.24.1
```
I'd stick to go1.23 for now; there's multiple regressions we ran into with go1.24, and they happen .. in non-obvious ways; 

- https://github.com/moby/moby/issues/49513
- https://github.com/dagger/dagger/issues/9759

milosgajdos replied:
I think having this in CI matrics is fine, but yeah, will change the Dockerfiles

thaJeztah replied:
Yeah, CI is ok for sure. It's indeed for the releases that I don't know if we'd hit it; more so because there's no clear reproducer yet, so users may hit it randomly, which isn't .. great.

milosgajdos replied:
OK dropped it back to 1.23.7 everywhere except the CI. PTAL

thaJeztah reviewed (approved):
LGTM

there's a typo in the last commit message though ☺️

milosgajdos commented:
> there's a typo in the last commit message though ☺️

I'm partially dyslexic so often miss these. Thanks, I've squashed and fixed the commit message.
